src/blockchain_db/sqlite/db_sqlite.h

```diff
@@ -0,0 +1,111 @@
+  bool add_sn_payments(std::vector<cryptonote::batch_sn_payment>& payments);
+  bool subtract_sn_payments(std::vector<cryptonote::batch_sn_payment>& payments);
+
+  // get_payments -> passing a block height will return an array of payments that should be created in a coinbase transaction on that block given the current batching DB state.
+  std::optional<std::vector<cryptonote::batch_sn_payment>> get_sn_payments(uint64_t block_height);
+
+  // calculate_rewards -> takes the list of contributors from sn_info with their SN contribution amounts and will calculate how much of the block rewards should be the allocated to the contributors. The function will return a list suitable for passing to add_sn_payments
+  std::vector<cryptonote::batch_sn_payment> calculate_rewards(uint8_t hf_version, uint64_t distribution_amount, service_nodes::service_node_info sn_info);
+
+  // add/pop_block -> takes a block that contains new block rewards to be batched and added to the database
+  // and/or batching payments that need to be subtracted from the database, in addition it takes a reference to
+  // the service node state which it will use to calculate the individual payouts.
+  // The function will then process this block add and subtracting to the batching DB appropriately.
+  // This is the primary entry point for the blockchain to add to the batching database.
+  // Each accepted block should call this passing in the SN list structure.
+  bool add_block(const cryptonote::block& block, const service_nodes::service_node_list::state_t& service_nodes_state);
+  bool pop_block(const cryptonote::block& block, const service_nodes::service_node_list::state_t& service_nodes_state);
+
+  // validate_batch_payment -> used to make sure that list of miner_tx_vouts is correct. Compares the miner_tx_vouts with a list previously extracted payments to make sure that the correct persons are being paid.
+  bool validate_batch_payment(std::vector<std::tuple<crypto::public_key, uint64_t>> miner_tx_vouts, std::vector<cryptonote::batch_sn_payment> calculated_payments_from_batching_db, uint64_t block_height);
+  
+  // these keep track of payments made to SN operators after then payment has been made. Allows for popping blocks back and knowing who got paid in those blocks.
+  // passing in a list of people to be marked as paid in the paid_amounts vector. Block height will be added to the batched_payments_paid database as height_paid.
+  bool save_payments(uint64_t block_height, std::vector<batch_sn_payment> paid_amounts);
+  std::vector<cryptonote::batch_sn_payment> get_block_payments(uint64_t block_height);
+  bool delete_block_payments(uint64_t block_height);
+
+  uint64_t height;
+
+protected:
+
+  cryptonote::network_type m_nettype;
+  std::string filename;
+
+};
+
+class BlockchainSQLiteTest : public BlockchainSQLite
+{
+public:
+  BlockchainSQLiteTest(cryptonote::network_type nettype, fs::path db_path)
+    : BlockchainSQLite(nettype, db_path) {};
```

Comment:
can we have this constructor declared only in the header and then implemented in a source file?